packages/eslint-plugin/src/rules/no-deprecated-components.ts

```diff
@@ -0,0 +1,172 @@
+/* !
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ */
+
+import { TSESTree } from "@typescript-eslint/utils";
+
+import { createRule } from "./utils/createRule";
+
+const DEPRECATED_TO_NEW_MAPPING: { [deprecated: string]: string } = {
+    AbstractComponent: "AbstractComponent2",
+    AbstractPureComponent: "AbstractPureComponent2",
+    CollapsibleList: "OverflowList",
+    MultiSelect: "MultiSelect2",
+    PanelStack: "PanelStack2",
+    Popover: "Popover2",
+    Select: "Select2",
+    Suggest: "Suggest2",
+    Tooltip: "Tooltip2",
+};
+const PACKAGES_WITH_DEPRECATED_IMPORTS = ["@blueprintjs/core", "@blueprintjs/select"];
+
+type MessageIds = "nonDeprecated" | "deprecated";
```

Comment:
I don't think we need two message IDs, if only one ("nonDeprecated") is used.

Reply:
ah yeah my original thought was if we had components that were removed with no replacement, but i didn't end up capturing those here